packages/amplify-category-function/src/index.ts

```diff
@@ -156,6 +161,11 @@ export async function initEnv(context) {
         _.set(teamProviderInfo, [envName, 'categories', categoryName, resourceName], tpiResourceParams);
         _.set(amplifyMeta, [categoryName, resourceName, 's3Bucket'], s3Bucket);
       }
+
+      // if the function has secrets, set the appId key in team-provider-info
+      if (!!getLocalFunctionSecretNames(resourceName, { fromCurrentCloudBackend: true }).length) {
```

Comment:
nit: I think it's more readable to compare the length, instead of `!!`
```suggestion
      if (getLocalFunctionSecretNames(resourceName, { fromCurrentCloudBackend: true }).length > 0) {
```

Reply:
```suggestion
     if (getLocalFunctionSecretNames(resourceName, { fromCurrentCloudBackend: true })?.length > 0) {
```

Reply:
getLocalFunctionSecretNames always returns an array (empty array if no secrets present) so I don't think optional access is necessary. Can change to `> 0` though